keyboards/handwired/yamsek/rules.mk

```diff
@@ -0,0 +1,31 @@
+#
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+
+CC_FLAGS += -Werror
+
+
+# override quantums matrix.c with custom matrix file
+#   which uses a row-oriented internal data-structure, we use a column oriented
+#   that was inherited from the I2C/MCP23018 code of the ergodox/ez
+CUSTOM_MATRIX = yes
```

Comment:
Although full custom matrix (`CUSTOM_MATRIX = yes`) is still supported, the preferred way to implement a custom matrix is `CUSTOM_MATRIX = lite`.  Please consider converting your matrix implementation to the “lite” variant (it would become much smaller, because you won't need to copy as much boilerplate code as you currently have there).

Reply:
the docs/custom_matrix could "encourage" the preferred way a bit more :P

waay back when i figured the whole port-expander part out, i went with a "too slightly" different data-order for the matrix: column-major instead of row-major -> matrix_row_t vs matrix_col_t; and the "lite" = quantum/matrix.c works with the row-major one

the col-major made (and still makes) more sens... so may i keep the fully custom matrix.c? (-: